tests/impls/descriptor_generator/test_prepopulated.py

```diff
@@ -0,0 +1,35 @@
+import unittest
+from smqtk_descriptors.impls.descriptor_generator.prepopulated import (
+    PrePopulatedDescriptorGenerator,
+)
+
+
+class TestPrePopulatedDescriptorGenerator(unittest.TestCase):
```

Comment:
I have a few comments about this.

1. I prefer the per-test comments to live in the docstring rather than a `#`-style comment above the function.
2. You have a `setUp` method that creates an instance of the generator, but then `test_generate_arrays_raises_assertion_error` creates its own instance? 
3. My opinion is that the `unittest` boilerplate is overkill here - perhaps generated by ChatGPT? You could rewrite this in a simplified pytest-style, where there are just the 3 test functions with names that start with `test_`, and each of the functions creates the instance of `generator` that it needs, so there is no reliance on the state of a unittest.TestCase instance.

WRT to point3, I'm not sure if @Purg has a different opinion on `unittest` vs `pytest` style tests. But I like the pattern where pytest scrapes each test file for any function where the function name starts with "test_", and then just runs them as normal functions. The setup/teardown boilerplate seems live overkill here.

Reply:
I would agree that the simpler "pytest"-style approach here is more appropriate (simplicity over complexity). I understand that a number of the tests here are written using `unittest.TestCase`, but that is more history than current preferences.

Side-bar: I might stay away, for now, from some of pytest's more implicit functionalities (like fixtures) unless you have a justified reason to use them. I view that unit-tests are allowed to be intentionally verbose or repetitive if only in the service of being highly explicit and understandable.

Reply:
I agree that fixtures are almost always not the way (I have very strong opinions on this, and writing the xdoctest pytest plugin was not fun). But I like using simple pytest `def test_<the-thing>` style tests as the standard. IMO: It's the right balance of reduced boilerplate / understandability.

Reply:
Modified unit test to use "pytest"-style